app/models/workflow_registry.rb

```diff
@@ -4,32 +4,17 @@
 #   provides data about workflows in agregate
 class WorkflowRegistry
   class EntryNotFound < StandardError; end
-
-  class_attribute :hash
-  self.hash = {}
-
-  def self.register(resource_class:, workflow_class:)
-    hash[resource_class] = workflow_class
-    true
-  end
-
-  def self.workflow_for(resource_class)
-    hash.fetch(resource_class)
-  rescue KeyError
-    raise EntryNotFound, resource_class
-  end
-
-  def self.unregister(resource_class)
-    hash.delete(resource_class)
-  end
-
   # @return array of strings
   def self.all_states
-    hash.values.uniq.map { |klass| klass.new(nil).valid_states }.flatten.uniq
+    workflows.map { |klass| klass.new(nil).valid_states }.flatten.uniq
   end
 
   # @return array of strings
   def self.public_read_states
-    hash.values.uniq.map(&:public_read_states).flatten.uniq
+    workflows.map(&:public_read_states).flatten.uniq
+  end
+
+  def self.workflows
+    [BoxWorkflow, FolderWorkflow, GeoWorkflow, BookWorkflow, DraftCompleteWorkflow]
```

Comment:
this seems like a good solution but I wonder if there's a way to dynamically generate this array. i'm worried it will be easy to miss this step when creating a new workflow, and there wouldn't be a test or obvious bug to catch the omission.

Reply:
`BaseWorkflow.descendants`?

Reply:
actually we could use that in a test just to make sure everything is registered if we don't want to use it in runtime.